local/kcp/start.sh

```diff
@@ -52,7 +52,7 @@ kcp-binaries() {
 
 kcp-start() {
   printf "Starting KCP server ...\n"
-  (cd "${KCP_RUNTIME_DIR}" && exec "${KCP_DIR}/bin/kcp" start "${PARAMS[@]}") &> "${KCP_RUNTIME_DIR}/kcp.log" &
+  (cd "${KCP_RUNTIME_DIR}" && exec "${KCP_DIR}/bin/kcp" start ${PARAMS}) &> "${KCP_RUNTIME_DIR}/kcp.log" &
```

Comment:
I think ${PARAMS} still can be quoted which is what the shellcheck is warning us about - **"${PARAMS}"**
(without the [@])

Reply:
my experience shows otherwise. I have you tested it?

Reply:
sorry for the delay. I tried running it locally. The script is actually running with the older code ("${PARAMS[@]}"). But I'm not sure on how to verify that its reading and processing those params.

./local/kcp/start.sh -v 6 --profiler-address localhost:6060 --discovery-poll-interval 3s

Snippet from logs:
```
Starting KCP server ...
KCP server started: 209332
Waiting for KCP to be ready ...
KCP ready
KUBECONFIG=/tmp/kcp-pipeline-service.fE347styk/.kcp/admin.kubeconfig
```

Reply:
`.start.sh` without any additional configuration works OK for me on both `main` and `pr-228`.

`PARAMS="--token-auth-file ../kcp-tokens --profiler-address localhost:6060 -v 6" ./start.sh` fails on both `main` and `pr-228`.

@fgiloux Could you provide us with the command that fails on `main` and succeeds in `pr-228`?

Reply:
~~~
$ KCP_RUNTIME_DIR=/tmp/kcp-runtime KCP_DIR=/home/fgiloux/development/kcp-dev/kcp PARAMS="--feature-gates=KCPSyncerTunnel=true --token-auth-file /tmp/kcp-tokens --profiler-address localhost:6060 -v 6" ./local/kcp/start.sh
~~~
works with PR-228, does not with main:
~~~
Error: invalid argument "KCPSyncerTunnel=true --token-auth-file /tmp/kcp-tokens --profiler-address localhost:6060 -v 6" for "--feature-gates" flag: invalid value of KCPSyncerTunnel=true --token-auth-file /tmp/kcp-tokens --profiler-address localhost:6060 -v 6, err: strconv.ParseBool: parsing "true --token-auth-file /tmp/kcp-tokens --profiler-address localhost:6060 -v 6": invalid syntax
~~~

It is not just that the command needs to succeed but the parameters need to be processed as well. You can check that with the token file for instance.
I thought that using brackets would do the trick with main and the server starts but the later parameters like the auth file are not taken in consideration.
~~~
$ KCP_RUNTIME_DIR=/tmp/kcp-runtime KCP_DIR=/home/fgiloux/development/kcp-dev/kcp PARAMS="(--feature-gates=KCPSyncerTunnel=true --token-auth-file /tmp/kcp-tokens --profiler-address localhost:6060 -v 6)" ./local/kcp/start.sh
~~~

The command you shared works for me with PR-228.

Reply:
If you use brackets, I'd think line 119 might need to be changed to `PARAMS="${PARAMS[@]:-}"` to retain all elements.

Reply:
@fgiloux I believe this is how to solve the issue without angering shellcheck:
``` 
PARAMS="${PARAMS:-"--token-auth-file ${PWD}/kcp-tokens --profiler-address localhost:6060 -v 2"}"
mapfile -d ' ' PARAMS < <(echo "$PARAMS")
```
First line sets the default value, the second one converts it into an array.

I use the following script to validate the behavior:
```
test() {
  # Set default value
  PARAMS="${PARAMS:-"123 abc"}"
  # Transform params into an array
  mapfile -d ' ' PARAMS < <(echo "$PARAMS")
  # Each word should be on its own line
  python -c "
import sys
for a in sys.argv[1:]:
  print(a)
" "${PARAMS[@]}"
}
test
PARAMS="$RANDOM foo bar" test
```

Reply:
@Roming22 you should try to run: `kcp start options`

Reply:
I've submitted a PR with the fix: see #235 .